var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -319,20 +469,22 @@ def cmake_args(self):
                 elif "intel" in self.spec.variants["backend"].value:
                     intel_lib = (
                         self.spec["intel-oneapi-compilers"].prefix
-                        + "/compiler/2023.0.0/linux/lib/libOpenCL.so"
+                        + "/compiler/"
+                        + str(self.spec["intel-oneapi-compilers"].version)
+                        + "/linux/lib/libOpenCL.so"
                     )
                     args.append("-DOpenCL_LIBRARY=" + intel_lib)
                 elif "pocl" in self.spec.variants["backend"].value:
-                    args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+                    args.append("-DCMAKE_CXX_COMPILER=" + spack_cxx)
                     pocl_lib = self.spec["pocl"].prefix + "/lib64/libOpenCL.so"
                     args.append("-DOpenCL_LIBRARY=" + pocl_lib)
-                args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+                args.append("-DCMAKE_CXX_COMPILER=" + spack_cxx)
 
         # ===================================
         #             RAJA
         # ===================================
         if "+raja" in self.spec:
-            args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+            args.append("-DCMAKE_CXX_COMPILER=" + spack_cxx)
```

Comment:
This seems to be set for every combination -- could we separate it out and actually set it unconditionally or am I missing something?